data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandlerTest.java

```diff
@@ -174,75 +186,91 @@ public void shouldReturnBlockWhenEpochQuery() throws Exception {
     final String epochNum = "1";
     final Map<String, List<String>> params = Map.of(EPOCH, List.of(epochNum));
 
-    when(context.queryParamMap()).thenReturn(params);
-    when(context.queryParam(any())).thenReturn(epochNum);
-    when(storageClient.getBlockRootBySlot(any())).thenReturn(Optional.of(blockRoot));
-    when(storageClient.getStore()).thenReturn(store);
-    when(store.getSignedBlock(any())).thenReturn(signedBeaconBlock);
-    when(historicalChainData.getFinalizedBlockAtSlot(any()))
-        .thenReturn(SafeFuture.completedFuture(empty()));
+    doReturn(params).when(context).queryParamMap();
```

Comment:
We prefer `when(context.queryParamMap()).thenReturn(...)` if at all possible because it's actually type safe. This includes if you have to declare the mock as a generic type and use `SuppressWarnings` because `mock(Supplier.class)` is untyped. Otherwise if the return type changes, the compiler won't flag it. 

But sometimes it's unavoidable  (e.g. if something returns a `SafeFuture<?>` you can't ever have something that satisfies that because it can't know what `?` is to see if you satisfied it, and I think there are a few other corner cases as well.